// react-form-wizard-component/dist/react-form-wizard-component.es.js
/*
  * react-form-wizard-component
  * Creator:Parsa Jiravand
  * A react form wizard component with validation and progress bar with no external depenendcies which simplifies tab wizard management.
  * v0.1.7
  * MIT License
  */
 
import z, { useImperativeHandle as lr, useState as cr, useEffect as $e } from "react";
var Ee = { exports: {} }, K = {};
/**
 * @license React
 * react-jsx-runtime.production.min.js
 *
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
var ze;
function fr() {
  if (ze)
    return K;
  ze = 1;
  var w = z, m = Symbol.for("react.element"), C = Symbol.for("react.fragment"), j = Object.prototype.hasOwnProperty, g = w.__SECRET_INTERNALS_DO_NOT_USE_OR_YOU_WILL_BE_FIRED.ReactCurrentOwner, O = { key: !0, ref: !0, __self: !0, __source: !0 };
  function F(k, p, N) {
    var d, E = {}, T = null, R = null;
    N !== void 0 && (T = "" + N), p.key !== void 0 && (T = "" + p.key), p.ref !== void 0 && (R = p.ref);
    for (d in p)
      j.call(p, d) && !O.hasOwnProperty(d) && (E[d] = p[d]);
    if (k && k.defaultProps)
      for (d in p = k.defaultProps, p)
        E[d] === void 0 && (E[d] = p[d]);
    return { $$typeof: m, type: k, key: T, ref: R, props: E, _owner: g.current };
  }
  return K.Fragment = C, K.jsx = F, K.jsxs = F, K;
}
var X = {};
/**
 * @license React
 * react-jsx-runtime.development.js
 *
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
var Ie;
function dr() {
  return Ie || (Ie = 1, process.env.NODE_ENV !== "production" && function() {
    var w = z, m = Symbol.for("react.element"), C = Symbol.for("react.portal"), j = Symbol.for("react.fragment"), g = Symbol.for("react.strict_mode"), O = Symbol.for("react.profiler"), F = Symbol.for("react.provider"), k = Symbol.for("react.context"), p = Symbol.for("react.forward_ref"), N = Symbol.for("react.suspense"), d = Symbol.for("react.suspense_list"), E = Symbol.for("react.memo"), T = Symbol.for("react.lazy"), R = Symbol.for("react.offscreen"), W = Symbol.iterator, Z = "@@iterator";
    function se(e) {
      if (e === null || typeof e != "object")
        return null;
      var r = W && e[W] || e[Z];
      return typeof r == "function" ? r : null;
    }
    var _ = w.__SECRET_INTERNALS_DO_NOT_USE_OR_YOU_WILL_BE_FIRED;
    function f(e) {
      {
        for (var r = arguments.length, t = new Array(r > 1 ? r - 1 : 0), n = 1; n < r; n++)
          t[n - 1] = arguments[n];
        y("error", e, t);
      }
    }
    function y(e, r, t) {
      {
        var n = _.ReactDebugCurrentFrame, u = n.getStackAddendum();
        u !== "" && (r += "%s", t = t.concat([u]));
        var c = t.map(function(o) {
          return String(o);
        });
        c.unshift("Warning: " + r), Function.prototype.apply.call(console[e], console, c);
      }
    }
    var I = !1, Q = !1, q = !1, B = !1, ee = !1, re;
    re = Symbol.for("react.module.reference");
    function ue(e) {
      return !!(typeof e == "string" || typeof e == "function" || e === j || e === O || ee || e === g || e === N || e === d || B || e === R || I || Q || q || typeof e == "object" && e !== null && (e.$$typeof === T || e.$$typeof === E || e.$$typeof === F || e.$$typeof === k || e.$$typeof === p || // This needs to include all possible module reference object
      // types supported by any Flight configuration anywhere since
      // we don't know which Flight build this will end up being used
      // with.
      e.$$typeof === re || e.getModuleId !== void 0));
    }
    function le(e, r, t) {
      var n = e.displayName;
      if (n)
        return n;
      var u = r.displayName || r.name || "";
      return u !== "" ? t + "(" + u + ")" : t;
    }
    function Y(e) {
      return e.displayName || "Context";
    }
    function P(e) {
      if (e == null)
        return null;
      if (typeof e.tag == "number" && f("Received an unexpected object in getComponentNameFromType(). This is likely a bug in React. Please file an issue."), typeof e == "function")
        return e.displayName || e.name || null;
      if (typeof e == "string")
        return e;
      switch (e) {
        case j:
          return "Fragment";
        case C:
          return "Portal";
        case O:
          return "Profiler";
        case g:
          return "StrictMode";
        case N:
          return "Suspense";
        case d:
          return "SuspenseList";
      }
      if (typeof e == "object")
        switch (e.$$typeof) {
          case k:
            var r = e;
            return Y(r) + ".Consumer";
          case F:
            var t = e;
            return Y(t._context) + ".Provider";
          case p:
            return le(e, e.render, "ForwardRef");
          case E:
            var n = e.displayName || null;
            return n !== null ? n : P(e.type) || "Memo";
          case T: {
            var u = e, c = u._payload, o = u._init;
            try {
              return P(o(c));
            } catch {
              return null;
            }
          }
        }
      return null;
    }
    var s = Object.assign, l = 0, x, D, A, V, L, J, G;
    function Re() {
    }
    Re.__reactDisabledLog = !0;
    function Ae() {
      {
        if (l === 0) {
          x = console.log, D = console.info, A = console.warn, V = console.error, L = console.group, J = console.groupCollapsed, G = console.groupEnd;
          var e = {
            configurable: !0,
            enumerable: !0,
            value: Re,
            writable: !0
          };
          Object.defineProperties(console, {
            info: e,
            log: e,
            warn: e,
            error: e,
            group: e,
            groupCollapsed: e,
            groupEnd: e
          });
        }
        l++;
      }
    }
    function We() {
      {
        if (l--, l === 0) {
          var e = {
            configurable: !0,
            enumerable: !0,
            writable: !0
          };
          Object.defineProperties(console, {
            log: s({}, e, {
              value: x
            }),
            info: s({}, e, {
              value: D
            }),
            warn: s({}, e, {
              value: A
            }),
            error: s({}, e, {
              value: V
            }),
            group: s({}, e, {
              value: L
            }),
            groupCollapsed: s({}, e, {
              value: J
            }),
            groupEnd: s({}, e, {
              value: G
            })
          });
        }
        l < 0 && f("disabledDepth fell below zero. This is a bug in React. Please file an issue.");
      }
    }
    var ce = _.ReactCurrentDispatcher, fe;
    function te(e, r, t) {
      {
        if (fe === void 0)
          try {
            throw Error();
          } catch (u) {
            var n = u.stack.trim().match(/\n( *(at )?)/);
            fe = n && n[1] || "";
          }
        return `
` + fe + e;
      }
    }
    var de = !1, ne;
    {
      var Ye = typeof WeakMap == "function" ? WeakMap : Map;
      ne = new Ye();
    }
    function _e(e, r) {
      if (!e || de)
        return "";
      {
        var t = ne.get(e);
        if (t !== void 0)
          return t;
      }
      var n;
      de = !0;
      var u = Error.prepareStackTrace;
      Error.prepareStackTrace = void 0;
      var c;
      c = ce.current, ce.current = null, Ae();
      try {
        if (r) {
          var o = function() {
            throw Error();
          };
          if (Object.defineProperty(o.prototype, "props", {
            set: function() {
              throw Error();
            }
          }), typeof Reflect == "object" && Reflect.construct) {
            try {
              Reflect.construct(o, []);
            } catch ($) {
              n = $;
            }
            Reflect.construct(e, [], o);
          } else {
            try {
              o.call();
            } catch ($) {
              n = $;
            }
            e.call(o.prototype);
          }
        } else {
          try {
            throw Error();
          } catch ($) {
            n = $;
          }
          e();
        }
      } catch ($) {
        if ($ && n && typeof $.stack == "string") {
          for (var a = $.stack.split(`
`), b = n.stack.split(`
`), v = a.length - 1, h = b.length - 1; v >= 1 && h >= 0 && a[v] !== b[h]; )
            h--;
          for (; v >= 1 && h >= 0; v--, h--)
            if (a[v] !== b[h]) {
              if (v !== 1 || h !== 1)
                do
                  if (v--, h--, h < 0 || a[v] !== b[h]) {
                    var S = `
` + a[v].replace(" at new ", " at ");
                    return e.displayName && S.includes("<anonymous>") && (S = S.replace("<anonymous>", e.displayName)), typeof e == "function" && ne.set(e, S), S;
                  }
                while (v >= 1 && h >= 0);
              break;
            }
        }
      } finally {
        de = !1, ce.current = c, We(), Error.prepareStackTrace = u;
      }
      var U = e ? e.displayName || e.name : "", De = U ? te(U) : "";
      return typeof e == "function" && ne.set(e, De), De;
    }
    function Ve(e, r, t) {
      return _e(e, !1);
    }
    function Le(e) {
      var r = e.prototype;
      return !!(r && r.isReactComponent);
    }
    function ae(e, r, t) {
      if (e == null)
        return "";
      if (typeof e == "function")
        return _e(e, Le(e));
      if (typeof e == "string")
        return te(e);
      switch (e) {
        case N:
          return te("Suspense");
        case d:
          return te("SuspenseList");
      }
      if (typeof e == "object")
        switch (e.$$typeof) {
          case p:
            return Ve(e.render);
          case E:
            return ae(e.type, r, t);
          case T: {
            var n = e, u = n._payload, c = n._init;
            try {
              return ae(c(u), r, t);
            } catch {
            }
          }
        }
      return "";
    }
    var ie = Object.prototype.hasOwnProperty, be = {}, Ce = _.ReactDebugCurrentFrame;
    function oe(e) {
      if (e) {
        var r = e._owner, t = ae(e.type, e._source, r ? r.type : null);
        Ce.setExtraStackFrame(t);
      } else
        Ce.setExtraStackFrame(null);
    }
    function Me(e, r, t, n, u) {
      {
        var c = Function.call.bind(ie);
        for (var o in e)
          if (c(e, o)) {
            var a = void 0;
            try {
              if (typeof e[o] != "function") {
                var b = Error((n || "React class") + ": " + t + " type `" + o + "` is invalid; it must be a function, usually from the `prop-types` package, but received `" + typeof e[o] + "`.This often happens because of typos such as `PropTypes.function` instead of `PropTypes.func`.");
                throw b.name = "Invariant Violation", b;
              }
              a = e[o](r, o, n, t, null, "SECRET_DO_NOT_PASS_THIS_OR_YOU_WILL_BE_FIRED");
            } catch (v) {
              a = v;
            }
            a && !(a instanceof Error) && (oe(u), f("%s: type specification of %s `%s` is invalid; the type checker function must return `null` or an `Error` but returned a %s. You may have forgotten to pass an argument to the type checker creator (arrayOf, instanceOf, objectOf, oneOf, oneOfType, and shape all require an argument).", n || "React class", t, o, typeof a), oe(null)), a instanceof Error && !(a.message in be) && (be[a.message] = !0, oe(u), f("Failed %s type: %s", t, a.message), oe(null));
          }
      }
    }
    var Ue = Array.isArray;
    function ve(e) {
      return Ue(e);
    }
    function qe(e) {
      {
        var r = typeof Symbol == "function" && Symbol.toStringTag, t = r && e[Symbol.toStringTag] || e.constructor.name || "Object";
        return t;
      }
    }
    function Be(e) {
      try {
        return we(e), !1;
      } catch {
        return !0;
      }
    }
    function we(e) {
      return "" + e;
    }
    function je(e) {
      if (Be(e))
        return f("The provided key is an unsupported type %s. This value must be coerced to a string before before using it here.", qe(e)), we(e);
    }
    var H = _.ReactCurrentOwner, Je = {
      key: !0,
      ref: !0,
      __self: !0,
      __source: !0
    }, Te, xe, pe;
    pe = {};
    function Ge(e) {
      if (ie.call(e, "ref")) {
        var r = Object.getOwnPropertyDescriptor(e, "ref").get;
        if (r && r.isReactWarning)
          return !1;
      }
      return e.ref !== void 0;
    }
    function He(e) {
      if (ie.call(e, "key")) {
        var r = Object.getOwnPropertyDescriptor(e, "key").get;
        if (r && r.isReactWarning)
          return !1;
      }
      return e.key !== void 0;
    }
    function Ke(e, r) {
      if (typeof e.ref == "string" && H.current && r && H.current.stateNode !== r) {
        var t = P(H.current.type);
        pe[t] || (f('Component "%s" contains the string ref "%s". Support for string refs will be removed in a future major release. This case cannot be automatically converted to an arrow function. We ask you to manually fix this case by using useRef() or createRef() instead. Learn more about using refs safely here: https://reactjs.org/link/strict-mode-string-ref', P(H.current.type), e.ref), pe[t] = !0);
      }
    }
    function Xe(e, r) {
      {
        var t = function() {
          Te || (Te = !0, f("%s: `key` is not a prop. Trying to access it will result in `undefined` being returned. If you need to access the same value within the child component, you should pass it as a different prop. (https://reactjs.org/link/special-props)", r));
        };
        t.isReactWarning = !0, Object.defineProperty(e, "key", {
          get: t,
          configurable: !0
        });
      }
    }
    function Ze(e, r) {
      {
        var t = function() {
          xe || (xe = !0, f("%s: `ref` is not a prop. Trying to access it will result in `undefined` being returned. If you need to access the same value within the child component, you should pass it as a different prop. (https://reactjs.org/link/special-props)", r));
        };
        t.isReactWarning = !0, Object.defineProperty(e, "ref", {
          get: t,
          configurable: !0
        });
      }
    }
    var Qe = function(e, r, t, n, u, c, o) {
      var a = {
        // This tag allows us to uniquely identify this as a React Element
        $$typeof: m,
        // Built-in properties that belong on the element
        type: e,
        key: r,
        ref: t,
        props: o,
        // Record the component responsible for creating this element.
        _owner: c
      };
      return a._store = {}, Object.defineProperty(a._store, "validated", {
        configurable: !1,
        enumerable: !1,
        writable: !0,
        value: !1
      }), Object.defineProperty(a, "_self", {
        configurable: !1,
        enumerable: !1,
        writable: !1,
        value: n
      }), Object.defineProperty(a, "_source", {
        configurable: !1,
        enumerable: !1,
        writable: !1,
        value: u
      }), Object.freeze && (Object.freeze(a.props), Object.freeze(a)), a;
    };
    function er(e, r, t, n, u) {
      {
        var c, o = {}, a = null, b = null;
        t !== void 0 && (je(t), a = "" + t), He(r) && (je(r.key), a = "" + r.key), Ge(r) && (b = r.ref, Ke(r, u));
        for (c in r)
          ie.call(r, c) && !Je.hasOwnProperty(c) && (o[c] = r[c]);
        if (e && e.defaultProps) {
          var v = e.defaultProps;
          for (c in v)
            o[c] === void 0 && (o[c] = v[c]);
        }
        if (a || b) {
          var h = typeof e == "function" ? e.displayName || e.name || "Unknown" : e;
          a && Xe(o, h), b && Ze(o, h);
        }
        return Qe(e, a, b, u, n, H.current, o);
      }
    }
    var he = _.ReactCurrentOwner, Se = _.ReactDebugCurrentFrame;
    function M(e) {
      if (e) {
        var r = e._owner, t = ae(e.type, e._source, r ? r.type : null);
        Se.setExtraStackFrame(t);
      } else
        Se.setExtraStackFrame(null);
    }
    var me;
    me = !1;
    function ge(e) {
      return typeof e == "object" && e !== null && e.$$typeof === m;
    }
    function Oe() {
      {
        if (he.current) {
          var e = P(he.current.type);
          if (e)
            return `

Check the render method of \`` + e + "`.";
        }
        return "";
      }
    }
    function rr(e) {
      {
        if (e !== void 0) {
          var r = e.fileName.replace(/^.*[\\\/]/, ""), t = e.lineNumber;
          return `

Check your code at ` + r + ":" + t + ".";
        }
        return "";
      }
    }
    var ke = {};
    function tr(e) {
      {
        var r = Oe();
        if (!r) {
          var t = typeof e == "string" ? e : e.displayName || e.name;
          t && (r = `

Check the top-level render call using <` + t + ">.");
        }
        return r;
      }
    }
    function Pe(e, r) {
      {
        if (!e._store || e._store.validated || e.key != null)
          return;
        e._store.validated = !0;
        var t = tr(r);
        if (ke[t])
          return;
        ke[t] = !0;
        var n = "";
        e && e._owner && e._owner !== he.current && (n = " It was passed a child from " + P(e._owner.type) + "."), M(e), f('Each child in a list should have a unique "key" prop.%s%s See https://reactjs.org/link/warning-keys for more information.', t, n), M(null);
      }
    }
    function Fe(e, r) {
      {
        if (typeof e != "object")
          return;
        if (ve(e))
          for (var t = 0; t < e.length; t++) {
            var n = e[t];
            ge(n) && Pe(n, r);
          }
        else if (ge(e))
          e._store && (e._store.validated = !0);
        else if (e) {
          var u = se(e);
          if (typeof u == "function" && u !== e.entries)
            for (var c = u.call(e), o; !(o = c.next()).done; )
              ge(o.value) && Pe(o.value, r);
        }
      }
    }
    function nr(e) {
      {
        var r = e.type;
        if (r == null || typeof r == "string")
          return;
        var t;
        if (typeof r == "function")
          t = r.propTypes;
        else if (typeof r == "object" && (r.$$typeof === p || // Note: Memo only checks outer props here.
        // Inner props are checked in the reconciler.
        r.$$typeof === E))
          t = r.propTypes;
        else
          return;
        if (t) {
          var n = P(r);
          Me(t, e.props, "prop", n, e);
        } else if (r.PropTypes !== void 0 && !me) {
          me = !0;
          var u = P(r);
          f("Component %s declared `PropTypes` instead of `propTypes`. Did you misspell the property assignment?", u || "Unknown");
        }
        typeof r.getDefaultProps == "function" && !r.getDefaultProps.isReactClassApproved && f("getDefaultProps is only used on classic React.createClass definitions. Use a static property named `defaultProps` instead.");
      }
    }
    function ar(e) {
      {
        for (var r = Object.keys(e.props), t = 0; t < r.length; t++) {
          var n = r[t];
          if (n !== "children" && n !== "key") {
            M(e), f("Invalid prop `%s` supplied to `React.Fragment`. React.Fragment can only have `key` and `children` props.", n), M(null);
            break;
          }
        }
        e.ref !== null && (M(e), f("Invalid attribute `ref` supplied to `React.Fragment`."), M(null));
      }
    }
    function Ne(e, r, t, n, u, c) {
      {
        var o = ue(e);
        if (!o) {
          var a = "";
          (e === void 0 || typeof e == "object" && e !== null && Object.keys(e).length === 0) && (a += " You likely forgot to export your component from the file it's defined in, or you might have mixed up default and named imports.");
          var b = rr(u);
          b ? a += b : a += Oe();
          var v;
          e === null ? v = "null" : ve(e) ? v = "array" : e !== void 0 && e.$$typeof === m ? (v = "<" + (P(e.type) || "Unknown") + " />", a = " Did you accidentally export a JSX literal instead of a component?") : v = typeof e, f("React.jsx: type is invalid -- expected a string (for built-in components) or a class/function (for composite components) but got: %s.%s", v, a);
        }
        var h = er(e, r, t, u, c);
        if (h == null)
          return h;
        if (o) {
          var S = r.children;
          if (S !== void 0)
            if (n)
              if (ve(S)) {
                for (var U = 0; U < S.length; U++)
                  Fe(S[U], e);
                Object.freeze && Object.freeze(S);
              } else
                f("React.jsx: Static children should always be an array. You are likely explicitly calling React.jsxs or React.jsxDEV. Use the Babel transform instead.");
            else
              Fe(S, e);
        }
        return e === j ? ar(h) : nr(h), h;
      }
    }
    function ir(e, r, t) {
      return Ne(e, r, t, !0);
    }
    function or(e, r, t) {
      return Ne(e, r, t, !1);
    }
    var sr = or, ur = ir;
    X.Fragment = j, X.jsx = sr, X.jsxs = ur;
  }()), X;
}
process.env.NODE_ENV === "production" ? Ee.exports = fr() : Ee.exports = dr();
var i = Ee.exports;
const vr = z.forwardRef(
  ({
    title: w,
    icon: m,
    shape: C,
    color: j = "#2196f3",
    isActive: g,
    index: O,
    onClick: F
  }, k) => {
    const p = g ? "active" : "", N = C === "square" ? "default" : "", [d, E] = z.useState(!1);
    z.useEffect(() => {
      g && E(!0);
    }, [g]);
    const T = () => {
      if (g && d)
        return { color: "white" };
      if (d)
        return { color: j };
    };
    z.useImperativeHandle(k, () => ({
      setChecked: (W) => {
        E(W);
      }
    }));
    const R = () => m ? typeof m == "string" ? /* @__PURE__ */ i.jsx("i", { className: m, style: T() }) : m : /* @__PURE__ */ i.jsx("span", { style: T(), children: O + 1 });
    return /* @__PURE__ */ i.jsx("li", { className: p, children: /* @__PURE__ */ i.jsxs(
      "a",
      {
        className: g ? "active" : "",
        style: { cursor: N },
        onClick: F,
        children: [
          /* @__PURE__ */ i.jsx(
            "div",
            {
              className: `wizard-icon-circle md ${d ? "checked" : ""} ${C === "square" ? "square_shape" : ""}`,
              role: "tab",
              tabIndex: g ? 0 : void 0,
              id: `step-${O}`,
              "aria-controls": `step-${O}`,
              "aria-disabled": g,
              "aria-selected": g,
              style: {
                borderColor: d ? j : ""
              },
              children: /* @__PURE__ */ i.jsx(
                "div",
                {
                  className: `wizard-icon-container ${C === "square" ? "square_shape" : ""}`,
                  style: {
                    backgroundColor: g ? j : ""
                  },
                  children: /* @__PURE__ */ i.jsx("span", { className: "wizard-icon", children: R() })
                }
              )
            }
          ),
          /* @__PURE__ */ i.jsx(
            "span",
            {
              className: `stepTitle ${g ? "active" : ""}`,
              style: {
                color: d ? j : "",
                marginTop: "8px"
              },
              children: w
            }
          )
        ]
      }
    ) }, O);
  }
), ye = ({ onClick: w, children: m }) => /* @__PURE__ */ i.jsx("button", { className: "wizard-btn", type: "button", onClick: w, children: m });
const pr = z.forwardRef(
  ({
    title: w,
    shape: m = "",
    color: C = "#2196f3",
    children: j,
    subtitle: g = "",
    nextButtonText: O = "Next",
    nextButtonTemplate: F,
    backButtonText: k = "Back",
    backButtonTemplate: p,
    finishButtonText: N = "Finish",
    finishButtonTemplate: d,
    stepSize: E = "md",
    layout: T = "horizontal",
    startIndex: R = 0,
    onComplete: W,
    onTabChange: Z
  }, se) => {
    const _ = z.Children.toArray(
      j
    ), f = _.map(() => z.useRef(null));
    lr(se, () => ({
      nextTab: () => {
        q();
      },
      prevTab: () => {
        B();
      },
      reset: () => {
        I(R), f.forEach((s, l) => {
          var x, D;
          R >= l ? (x = s == null ? void 0 : s.current) == null || x.setChecked(!0) : (D = s == null ? void 0 : s.current) == null || D.setChecked(!1);
        });
      },
      activeAll: () => {
        f.forEach((s) => {
          var l;
          (l = s == null ? void 0 : s.current) == null || l.setChecked(!0);
        });
      },
      goToTab: (s) => {
        Q(s, !0), f.forEach((l, x) => {
          var D, A;
          s >= x ? (D = l == null ? void 0 : l.current) == null || D.setChecked(!0) : (A = l == null ? void 0 : l.current) == null || A.setChecked(!1);
        });
      }
    })), (R < 0 || R > _.length) && (R = 0, console.error(
      "startIndex should be greater than or equal to 0 or less than steps.length"
    ));
    const [y, I] = cr(R);
    $e(() => {
      y > 0 && f.forEach((s, l) => {
        var x;
        R >= l && ((x = s == null ? void 0 : s.current) == null || x.setChecked(!0));
      });
    }, [y, R, f]), typeof Z == "function" && Z({
      prevIndex: y,
      nextIndex: y + 1
    });
    const Q = (s, l = !1) => {
      if (l) {
        I(s);
        return;
      }
      s <= y && I(s);
    }, q = () => {
      y !== _.length - 1 && I(y + 1);
    }, B = () => {
      y !== 0 && I(y - 1);
    }, ee = () => {
      typeof W == "function" && W();
    }, re = () => _.map((s, l) => {
      const { title: x, icon: D, isValid: A = !0, validationError: V } = s.props, L = l === y;
      return $e(() => {
        var J, G;
        L && !A && (I(l - 1), (G = (J = f[l]) == null ? void 0 : J.current) == null || G.setChecked(!1), typeof V == "function" && V());
      }, [L, A, l, V]), /* @__PURE__ */ i.jsx(
        vr,
        {
          ref: f[l],
          title: x,
          icon: D,
          shape: m,
          color: C,
          isActive: L,
          index: l,
          onClick: () => Q(l)
        },
        l
      );
    }), ue = () => _[y], le = {
      backgroundColor: C,
      width: `${(y + 1) / _.length * 100}%`,
      color: C
    }, Y = {
      backgroundColor: C,
      borderColor: C,
      borderRadius: "4px"
    }, P = T === "vertical" ? "vertical" : "horizontal";
    return /* @__PURE__ */ i.jsxs("div", { className: `react-form-wizard ${E} ${P}`, children: [
      /* @__PURE__ */ i.jsx("div", { className: "wizard-header", children: typeof w == "string" ? /* @__PURE__ */ i.jsxs(i.Fragment, { children: [
        /* @__PURE__ */ i.jsx("h4", { className: "wizard-title", children: w }),
        /* @__PURE__ */ i.jsx("p", { className: "category", children: g })
      ] }) : w }),
      /* @__PURE__ */ i.jsxs("div", { className: "wizard-navigation", children: [
        /* @__PURE__ */ i.jsx("div", { className: "wizard-progress-with-circle", children: /* @__PURE__ */ i.jsx("div", { className: "wizard-progress-bar", style: le }) }),
        /* @__PURE__ */ i.jsx(
          "ul",
          {
            className: `form-wizard-steps  wizard-nav wizard-nav-pills ${m} ${E}`,
            style: { borderColor: C },
            children: re()
          }
        ),
        /* @__PURE__ */ i.jsx("div", { className: "wizard-tab-content", children: ue() })
      ] }),
      /* @__PURE__ */ i.jsxs("div", { className: "wizard-card-footer clearfix", children: [
        y > 0 && /* @__PURE__ */ i.jsx(i.Fragment, { children: p ? p(B) : /* @__PURE__ */ i.jsx("div", { className: "wizard-footer-left", style: Y, children: /* @__PURE__ */ i.jsx(ye, { onClick: B, children: k }) }) }),
        y < _.length - 1 && /* @__PURE__ */ i.jsx(i.Fragment, { children: F ? F(q) : /* @__PURE__ */ i.jsx("div", { className: "wizard-footer-right", style: Y, children: /* @__PURE__ */ i.jsx(ye, { onClick: q, children: O }) }) }),
        y === _.length - 1 && /* @__PURE__ */ i.jsx(i.Fragment, { children: d ? d(ee) : /* @__PURE__ */ i.jsx("div", { className: "wizard-footer-right", style: Y, children: /* @__PURE__ */ i.jsx(ye, { onClick: ee, children: N }) }) })
      ] })
    ] });
  }
);
pr.TabContent = ({ children: w, isValid: m = !0 }) => /* @__PURE__ */ i.jsx(i.Fragment, { children: m && w });
export {
  pr as default
};
